### ISSUE TITLE
add_coauthors.function doesn't work, when user_login has spaces?

### ISSUE THREAD

**Issue by TylerDurdon** [Automattic/Co-Authors-Plus#449]
Hi, really like the plugin, but users with space in user_login will not be added when using
$coauthors_plus->add_coauthors.
Was there a workaround or is there a fix?

Thx & Regards

**Comment by TylerDurdon** [comment]
is there anyone supporting this plugin?

**Comment by rebeccahum** [comment]
Hi @TylerDurdon, can you give me more specific steps to reproduce this?  I'm seeing that it's possible to add users with a space in their user_login to a post when I type their name in the box under "Authors".

**Comment by rebeccahum** [comment]
Hi @TylerDurdon, can you give me more specific steps to reproduce this?  I'm not sure what instance of `add_coauthors` you are experiencing this issue with.

**Comment by rebeccahum** [comment]
I'm seeing that when I delete a user and try to re-assign the posts to a user with spaces in their `user_login`, it doesn't propagate as expected.  

There are probably more instances of this happening with `add_coauthors`.

**Comment by rebeccahum** [comment]
I'm also seeing when you login as a user with spaces in their `user_login`, when you hit Posts > Add New, you receive the following the error message: 

```
Notice: Trying to get property of non-object in /wp-content/plugins/Co-Authors-Plus/co-authors-plus.php on line 841

Notice: Trying to get property of non-object in /wp-content/plugins/Co-Authors-Plus/co-authors-plus.php on line 851
```

**Comment by TylerDurdon** [comment]
HI rebeccahum,

when i add the user in a page/post > author it works perfect.
When i add the user with the function in php:

> add_coauthors( $this->author1_post1, array( $editor1->user_login ), true );

it doens't work, when the user has a space in his user_login name.

**Comment by TylerDurdon** [comment]
Hi @rebeccahum, i tried user-login and user_nicename, but there is no differnce in behaviour.
I always get the same users in the authors list after saving the site (using hook edit_form_after_title)
Without a space in user_login OR without "-" in user_nicename.

$coauthors_plus->add_coauthors($pageid , array($user->user_login), true); 
and
$coauthors_plus->add_coauthors($pageid , array($user->user_nicename), true); 

i dump my array, so the users to be added are clear to me, and the four users have the data:
1. user_login: admin 
user_nicename: admin
state: Works

2. user_login: Test User
user_nicename: test-user
state: Don't work

3. user_login: tester2
user_nicename: tester2
state: Works

4. user_login: torsten tester
user_nicename: torsten-tester
state: Don't work

after saving i have the authors admin and tester2 in the author-area in edit mode. any further idea?

**Comment by rebeccahum** [comment]
@TylerDurdon are you able to give me the snippet of code you are using?

**Comment by rebeccahum** [comment]
@TylerDurdon are you able to provide the snippet of code you are using and describe how the code is supposed to work versus how it's actually working?

**Comment by TylerDurdon** [comment]
@rebeccahum 

function add_authors() {
	global $coauthors_plus;

/* first i check the the page and the author */
        $page = get_post();
	$pageid = $page ->ID;
	$author = wp_get_current_user();
	$authorid = $author ->ID;

/* then i select all users with the same role as the author, cause i want all this users as authors */
	$user_info = get_userdata($authorid);
	$role = $user_info->roles;
	$args = array(
		'role' => $role[0],
	 ); 
	$rollenuser = get_users( $args );

 /* then i try to add them as co-authors*/

	foreach ($roleuser as $user) {
		if ($user != $author)
		{
		$coauthors_plus->add_coauthors($pageid , array($user->user_login), true); 
		}
	}
}

/* and the hook is */
add_action('edit_form_after_title', 'add_authors');


[ Wordpress 4.8.2 / CoAuthor Plus 3.2.2]

**Comment by rebeccahum** [comment]
Thanks @TylerDurdon.  Are you able to specify where you are placing this?  Also, what are the results are you getting versus the results you expect?

Thanks!

**Comment by rebeccahum** [comment]
Thanks @TylerDurdon.  Are you able to specify where you are placing this?  Also, what are the results are you getting versus the results you expect?

Thanks!

**Comment by rebeccahum** [comment]
Thanks @TylerDurdon. Are you able to specify where you are placing this? Also, what are the results are you getting versus the results you expect?

**Comment by philipjohn** [comment]
Fixed in #453
